File: src/main/solved/Programmers/level1/java/콜라츠 추측.java
package Programmers.level1.java;

class Solution {
    public int solution(int num) {
        int answer = 0;
        long result = (long)num;
        while(answer <= 500){
            if(result == 1) return answer;
            result = (result%2==0) ? result/=2 : result*3+1;
            answer+=1;
        }
        return -1;
    }
}

File: src/main/solved/Programmers/level1/kotlin/콜라츠 추측.kt
class Solution {
    fun solution(num: Int): Int {
        var n : Long = num.toLong()
        var answer = 0
        while(n != 1L){
            if (answer == 500){
                return -1;
            }

            if (n % 2 == 0L) {
                n /= 2;
            } else {
                n = n * 3 + 1;
            }
            answer += 1
        }

        return answer
    }
}